pkg/scheduler/algorithm/generic.go

```diff
@@ -103,16 +104,23 @@ func (g *genericScheduler) Schedule(ctx context.Context, fwk framework.Framework
 
 // selectClusters takes a prioritized list of clusters and then picks a fraction of clusters
 // in a reservoir sampling manner from the clusters that had the highest score.
-func (g *genericScheduler) selectClusters(clusterScoreList framework.ClusterScoreList, _ *appsapi.Subscription) ([]string, error) {
+func (g *genericScheduler) selectClusters(clusterScoreList framework.ClusterScoreList, sub *appsapi.Subscription, finv *appsapi.FeedInventory) (framework.TargetClusters, error) {
 	if len(clusterScoreList) == 0 {
-		return nil, fmt.Errorf("empty clusterScoreList")
+		return framework.TargetClusters{}, fmt.Errorf("empty clusterScoreList")
 	}
 
-	var selected []string
+	var selected framework.TargetClusters
 	for _, clusterScore := range clusterScoreList {
 		// TODO: sampling with scores
-		selected = append(selected, clusterScore.NamespacedName)
+		selected.BindingClusters = append(selected.BindingClusters, clusterScore.NamespacedName)
 	}
+
+	if sub.Spec.SchedulingStrategy == appsapi.DividingSchedulingStrategyType {
+		if err := g.divideReplicas(&selected, sub, finv); err != nil {
```

Comment:
`selected` will keep redundant clusters that are not bound for dividing scheduling.

Leave a knot here.